prometheus/promhttp/http.go

```diff
@@ -62,18 +62,22 @@ var gzipPool = sync.Pool{
 // no error logging, and it applies compression if requested by the client.
 //
 // The returned http.Handler is already instrumented using the
-// InstrumentMetricHandler function and the prometheus.DefaultRegisterer. If you
-// create multiple http.Handlers by separate calls of the Handler function, the
-// metrics used for instrumentation will be shared between them, providing
-// global scrape counts.
+// InstrumentMetricHandler function and the prometheus.DefaultRegisterer. The
+// latter is also used for the Registry field in HandlerOpts. If you create
+// multiple http.Handlers by separate calls of the Handler function, the metrics
+// used for instrumentation will be shared between them, providing global scrape
+// counts.
 //
 // This function is meant to cover the bulk of basic use cases. If you are doing
 // anything that requires more customization (including using a non-default
 // Gatherer, different instrumentation, and non-default HandlerOpts), use the
 // HandlerFor function. See there for details.
 func Handler() http.Handler {
 	return InstrumentMetricHandler(
-		prometheus.DefaultRegisterer, HandlerFor(prometheus.DefaultGatherer, HandlerOpts{}),
+		prometheus.DefaultRegisterer, HandlerFor(
+			prometheus.DefaultGatherer,
+			HandlerOpts{Registry: prometheus.DefaultRegisterer},
```

Comment:
I'm not sure this makes sense for the default handler, as gathering will fail the whole scrape anyway and once we've OpenMetrics it has an explicit EOF that'll catch encoding issues. Given that a little wary of adding two more series for ~every target, for cases that don't/won't happen.

Reply:
That's a good point. 99% of the usage for the new metric will be with `ContinueOnError`. However, the incident that triggered #542 would happen even with the default handler.

But yeah, adding metrics for the default case might be confusing. I'll remove it here.